mod/state-transition/pkg/core/state_processor_staking.go

```diff
@@ -118,22 +118,28 @@ func (sp *StateProcessor[
 ) error {
 	idx, err := st.ValidatorIndexByPubkey(dep.GetPubkey())
 	// If the validator already exists, we update the balance.
-	if err == nil {
-		var val ValidatorT
-		val, err = st.ValidatorByIndex(idx)
-		if err != nil {
-			return err
-		}
+	if err != nil {
+		// If the validator does not exist, we add the validator.
+		// Add the validator to the registry.
+		return sp.createValidator(st, dep)
+	}
 
-		// TODO: Modify balance here and then effective balance once per epoch.
-		val.SetEffectiveBalance(min(val.GetEffectiveBalance()+dep.GetAmount(),
-			math.Gwei(sp.cs.MaxEffectiveBalance())))
-		return st.UpdateValidatorAtIndex(idx, val)
+	var val ValidatorT
+	val, err = st.ValidatorByIndex(idx)
+	if err != nil {
+		return err
 	}
 
-	// If the validator does not exist, we add the validator.
-	// Add the validator to the registry.
-	return sp.createValidator(st, dep)
+	// TODO: Modify balance here and then effective balance once per epoch.
```

Comment:
can you remind me why we cant just set the balance here right now? And where would the effective balance code go, I assume in `processEpoch` ?

Reply:
@calbera,  eth specs would indeed update effective balance once per epch (see [this](https://github.com/ethereum/consensus-specs/blob/dev/specs/phase0/beacon-chain.md#effective-balances-updates)).
We currently do this every block, not sure why.
I would defer the update to epoch processing to a separate PR, if you agree